### PR TITLE
BU Arts Initiative: Profile Page (profile photo upload, event registration)

### PR DESCRIPTION
**Pull request by lcfmarco**
- Photo upload logic
- Registered events in explore page added to user schema
- Registered events showcased in profile page

### CODE REVIEW

File: bu_passport/lib/pages/profile_page.dart [hunk 1]
```diff
@@ -215,37 +201,31 @@ class _ProfilePageState extends State<ProfilePage>
                     if (selectedImageFile != null) ...[
                       Image.file(selectedImageFile!),
                       SizedBox(height: 8),
```
Comment (molly-yan) [inline bu_passport/lib/pages/profile_page.dart]: Replace magic number with dynamic measurement

File: bu_passport/lib/pages/profile_page.dart [hunk 2]
```diff
@@ -215,37 +201,31 @@ class _ProfilePageState extends State<ProfilePage>
...
+                                _image!) // Use MemoryImage if _image is not null
+                            : (userProfileImageUrl != null
+                                    ? NetworkImage(
+                                        userProfileImageUrl!) // Use the network image if available
+                                    : NetworkImage(
+                                        'https://via.placeholder.com/150') // Default image
+                                ) as ImageProvider,
+                      ),
                     ),
                     SizedBox(height: 8),
```
Comment (molly-yan) [inline bu_passport/lib/pages/profile_page.dart]: Replace magic number with dynamic measurement

File: bu_passport/lib/pages/profile_page.dart [hunk 3]
```diff
@@ -215,37 +201,31 @@ class _ProfilePageState extends State<ProfilePage>
...
-                    ElevatedButton(
-                        onPressed: _handleImageSelection,
-                        child: Text('Upload Profile Picture')),
-
+                    // ElevatedButton(
```
Comment (molly-yan) [inline bu_passport/lib/pages/profile_page.dart]: Removed unused code piece

**Review by molly-yan — CHANGES_REQUESTED** [review]
Good work! Just a few places need adjustments:
- Replace magic numbers with dynamic measurement.
- Removed unused code piece

**Comment by lcfmarco** [comment]
I just updated the file with the requested changes, pushed onto the profile page branch (I assume that is what I am meant to do with the requested changes? So hopefully it's all good and that you see it!)

**Review by molly-yan — APPROVED** [review]
lgtm. Please resolve the conflict, then you're good to merge.

**Pull request merged**